mslib/msui/wms_control.py

```diff
@@ -718,13 +894,50 @@ def clear_map(self):
         logging.debug("enabling checkboxes in map-options if any")
         self.signal_enable_cbs.emit()
 
-    def get_capabilities(self):
+    def select_layer_and_style(self, layer_list, layer_name, style_name):
+        if layer_list is None or layer_name is None:
+            return
+        self.current_sel_layer = None
+
+        for index in range(layer_list.topLevelItemCount()):
+            top_item = layer_list.topLevelItem(index)
+
+            # Iterate over the children of the top-level item
+            for i in range(top_item.childCount()):
+                child_item = top_item.child(i)
+
+                for column in range(child_item.columnCount()):
+
+                    if child_item.text(column).endswith(layer_name):
+                        self.current_sel_layer = child_item
+
+        if not self.current_sel_layer:
+            print(f"Layer '{layer_name}' not found.")
```

Comment:
when that is important only a popup reaches the user, needs to become checked later